src/serialbox-fortran/m_serialize.f90

```diff
@@ -1209,7 +1209,7 @@ SUBROUTINE fs_write_bool_0d(serializer, savepoint, fieldname, field)
                        istride, jstride, kstride, lstride)
   CALL fs_write_field_(serializer%serializer_ptr, savepoint%savepoint_ptr, &
                         TRIM(fieldname)//C_NULL_CHAR, &
-                       C_LOC(padd), istride, jstride, kstride, lstride)
+                       C_LOC(padd), istride, -1, -1, -1)
```

Comment:
I will this very counter-intuitive. Having a stride of -1 means that changing that corresponding index will actually change the memory location accessed. On the other hand, a stride of 0 exactly means that you can move to any position in that direction and it will never change the value you are accessing.

Since the fortran API always exposes 4 dimensions, there must be a way for the user to say "I'm not using that dimension", and I think that a stride of 0 has precisely this meaning, so we should stick to this.

Reply:
Our reason was to distinguish a missing dimension from a dimension with size 1. The compute strides will return `0` in case we have a dimension with size 1 which makes sense (https://github.com/eth-cscs/serialbox2/blob/7b68ba049608d90c8d1888ab2f63faf47bc37d24/src/serialbox-c/FortranWrapper.cpp#L145). But we cannot filter this dimension here: https://github.com/eth-cscs/serialbox2/pull/66/files#diff-e90743ed0072edf8cef89ddbbfae4225R56
That's why we decided to go for the `-1`. Does it make sense? I don't say it is a very good API but maybe good enough without having to change to much with unforeseen consequences.